### PR TITLE
adds cloudwatch tracking for contributions payments

### PR DESCRIPTION
**Pull request by michaelwmcnamara**
## Why are you doing this?
This is so that we can start to track payment attempt vs completion on contributions

## Changes
Add cloudwatch metrics for payment attempt, payment success and payment failure

### CODE REVIEW

File: frontend/app/monitoring/ContributorMetrics.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import com.amazonaws.services.cloudwatch.model.Dimension
+import com.gu.memsub.subsv2.SubscriptionPlan
+import controllers.Subscription
+import model.{ContributionPlanChoice, PlanChoice}
+
+class ContributorMetrics(val backendEnv: String) extends TouchpointBackendMetrics {
+
+  val service = "Contributor"
+
+  def putAttemptedSignUp: Unit ={
```
Comment (davidfurey) [inline frontend/app/monitoring/ContributorMetrics.scala]: Minor point, but style here is to put a space between `=` and `{` so:
`def putAttemptedSignUp: Unit = {`

Comment (michaelwmcnamara) [inline frontend/app/monitoring/ContributorMetrics.scala]: Thanks David - I will add that in :-)

File: frontend/app/monitoring/ContributorMetrics.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import com.amazonaws.services.cloudwatch.model.Dimension
+import com.gu.memsub.subsv2.SubscriptionPlan
+import controllers.Subscription
+import model.{ContributionPlanChoice, PlanChoice}
+
+class ContributorMetrics(val backendEnv: String) extends TouchpointBackendMetrics {
+
+  val service = "Contributor"
+
+  def putAttemptedSignUp: Unit = {
```
Comment (johnduffell) [inline frontend/app/monitoring/ContributorMetrics.scala]: if only one statement in a block, you'd normally omit the braces (no need to do another PR just for that though!)

**Comment by rtyley** [comment]
You hit 10000 commits on membership-frontend when you merged this!

image

**Comment by Ap0c** [comment]
> You hit 10000 commits on membership-frontend when you merged this!

🌈 🌟 🍦
